engine/src/main/java/io/zeebe/engine/nwe/container/ProcessProcessor.java

```diff
@@ -22,69 +26,173 @@
 
-    if (flowScopeContext.getIntent() == WorkflowInstanceIntent.ELEMENT_TERMINATING
-        && stateBehavior.isLastActiveExecutionPathInScope(childContext)) {
-      stateTransitionBehavior.transitionToTerminated(flowScopeContext);
+    boolean triggeredByEvent = false;
+    if (element.hasMessageStartEvent() || element.hasTimerStartEvent()) {
 
-    } else {
-      eventSubscriptionBehavior.publishTriggeredEventSubProcess(flowScopeContext);
+      triggeredByEvent = eventSubscriptionBehavior.publishTriggeredStartEvent(context);
     }
-  }
 
-  @Override
-  public Class<ExecutableFlowElementContainer> getType() {
-    return ExecutableFlowElementContainer.class;
-  }
+    if (!triggeredByEvent) {
 
-  @Override
-  public void onActivating(
-      final ExecutableFlowElementContainer element, final BpmnElementContext context) {}
+      final var noneStartEvent = element.getNoneStartEvent();
+      if (noneStartEvent == null) {
+        throw new BpmnProcessingException(
+            context, "Expected to activate the none start event of the process but not found.");
+      }
 
-  @Override
-  public void onActivated(
-      final ExecutableFlowElementContainer element, final BpmnElementContext context) {}
+      stateTransitionBehavior.activateChildInstance(context, noneStartEvent);
+    }
+  }
 
   @Override
   public void onCompleting(
-      final ExecutableFlowElementContainer element, final BpmnElementContext context) {}
+      final ExecutableFlowElementContainer element, final BpmnElementContext context) {
+
+    eventSubscriptionBehavior.unsubscribeFromEvents(context);
+    stateTransitionBehavior.transitionToCompleted(context);
+  }
 
   @Override
   public void onCompleted(
-      final ExecutableFlowElementContainer element, final BpmnElementContext context) {}
+      final ExecutableFlowElementContainer element, final BpmnElementContext context) {
+
+    final var parentWorkflowInstanceKey = context.getParentWorkflowInstanceKey();
+    final var parentElementInstanceKey = context.getParentElementInstanceKey();
+
+    if (parentWorkflowInstanceKey > 0) {
+      // workflow instance is created by a call activity
+
+      // TODO (saig0): move responsibility to call activity (#4473)
+      final var parentElementInstance = stateBehavior.getParentElementInstanceContext(context);
+
+      if (parentElementInstance != null
+          && parentElementInstance.getIntent() == WorkflowInstanceIntent.ELEMENT_ACTIVATED) {
+        // complete the corresponding call activity
+
+        stateTransitionBehavior.transitionToCompleting(parentElementInstance);
+
+        // propagate the variables to the parent
+        final var variablesState = stateBehavior.getVariablesState();
+
+        final var variables =
+            variablesState.getVariablesAsDocument(context.getElementInstanceKey());
+        variablesState.setTemporaryVariables(parentElementInstanceKey, variables);
+      }
+    }
+
+    if (element.hasNoneStartEvent()) {
```

Comment:
Is this check really enough to know that it was started as such?

Reply:
No, it just checks that the workflow has a none start event. The result sender itself checks if there is a pending request. This check prevents a state access for workflows that have only message/timer start events.